earth_moving/clutter/engine.py

```diff
@@ -207,18 +220,20 @@ def control_all_joints(self, joint_targets, targetVel, control_mode=p.VELOCITY_C
                     print("Warning: Iteration limit reached")
                         
         # if we are in position control, we just give the position
-        elif control_mode == p.POSITION_CONTROL:
-            for i in range(Njoints):
-                self.control_joint(i, joint_targets[i], control_mode,max_force, targetVel[i])
+        elif control_mode == p.POSITION_CONTROL:            
+            
+            # control
+            self.control_joint(joint_targets, control_mode)
                 
             # simulate
             self.simulate()
+            
         # something went wrong
         else:
             print("Error: Control mode not recognized")
 
     # give a set of waypoints, control the robot to reach them
-    def control_waypoints(self, waypoints, targetVel, control_mode=p.VELOCITY_CONTROL, max_force=1000):
+    def control_waypoints(self, waypoints, control_mode=p.VELOCITY_CONTROL):
```

Comment:
Type hinting is something worth considering (not only here).
For example, this line would be written as follows:
`def control_waypoints(self, waypoints: List[Tuple], ...`

Reply:
Didn't know about this, cool